onnxmltools/convert/sklearn/shape_calculators/LinearClassifier.py

```diff
@@ -36,26 +36,28 @@ def calculate_sklearn_linear_classifier_output_shapes(operator):
         class_labels = np.concatenate(class_labels)
     if all(isinstance(i, (six.string_types, six.text_type)) for i in class_labels):
         operator.outputs[0].type = StringTensorType(shape=[N])
-        if len(class_labels) > 2 or operator.type != 'SklearnLinearSVC':
-            # For multi-class classifier, we produce a map for encoding the probabilities of all classes
-            if operator.targeted_onnx_version < StrictVersion('1.2'):
-                operator.outputs[1].type = DictionaryType(StringTensorType([1]), FloatTensorType([1]))
+        if operator.type != 'SklearnKNeighborsClassifier' and 'NB' not in operator.type:
```

Comment:
Did you change the shape calculation rule described right below this function's header? This shape calculator requires that your classifier to have [N, C] input shape, so you need to accept [N, C] and reshape it into [C] where N is always 1 in your graph. Also, there are always two outputs, [one for label](http://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.BernoulliNB.html#sklearn.naive_bayes.BernoulliNB.predict) and the other [one for probability](http://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.BernoulliNB.html#sklearn.naive_bayes.BernoulliNB.predict_proba). Consequently, I don't think the shape calculation rules should be changed here.
Note that an important concept is that shape calculator determines the input and output shape of your graph, so the associated converter should make sure the produced sub-graph meets the shapes produced here.